grails-app/controllers/com/unifina/controller/api/KeyApiController.groovy

```diff
@@ -89,8 +95,14 @@ class KeyApiController {
 				permissionService.grant(request.apiUser, res, key, operation, false)
 
 				// If granting write, grant also read
-				if (operation == Permission.Operation.WRITE) {
-					permissionService.grant(request.apiUser, res, key, Permission.Operation.READ, false)
+				if (Stream.isAssignableFrom(params.resourceClass)) {
+					if (operation == Permission.Operation.STREAM_EDIT) {
```

Comment:
Ok some magic hacks here. Well, as I said, we'll get rid of these soon. Let's try the following:

(Let's instruct the frontend guys to send the `permission` as either `stream_publish` or `stream_subscribe`)

```
// Throws if user is not permitted to grant
permissionService.grant(request.apiUser, res, key, operation, false)

// Grant extra permissions depending on whether we're creating a publish or subscribe key
if (operation == Permission.Operation.STREAM_PUBLISH) {
    // grant also STREAM_GET and STREAM_SUBSCRIBE
} else if (operation == Permission.Operation.SUBSCRIBE) {
    // grant also STREAM_GET
} else {
    // invalid request
}
```